public-ui/src/app/data-browser/views/survey-chart/survey-chart-react.component.tsx

```diff
@@ -0,0 +1,182 @@
+    return surveyName + ' - Q' +  question.actualQuestionNumber + ' - ' +
+    question.conceptName + ' - ' + answer.stratum4 + ' - ' + g;
+  }
+
+
+  render() {
+      const {graphButtons, question, answer, searchTerm, surveyCountAnalysis, selectedResult, versionAnalysis} = this.props;
+      const { isLoaded, graphToShow, displayGraphErrorMessage, selectedChartAnalysis } = this.state;
+      const tabIndex = 0;
+      return <React.Fragment>
+        <div className='survey-graph-menu'>
+            {
+              graphButtons.map((g, index) => {
+                return (
+                 <div onClick={() => this.selectGraphType(g, question, answer)}
+                 className={graphToShow === g ? 'active survey-chart-choice' : 'survey-chart-choice'}
+                 tabIndex={tabIndex} key={index}>
+                 <span>{g}</span>
+                 <TooltipReactComponent tooltipKey={this.getTooltipKey(g)}
+                 label={this.getLabel(g)} searchTerm={searchTerm} action='Survey Chart Tooltip'>
+                 </TooltipReactComponent>
+                 </div>
+                 );
+              })
+            }
+        </div>
+{displayGraphErrorMessage
+    ? <div className='graph-error-message'>
+                  <ErrorMessageReactComponent dataType='chart'/>
+                </div>
+    : isLoaded && selectedChartAnalysis.analysisId === 3111 ?
+            <div className='chart' key='biosex-chart'>
+             <BioSexChartReactComponent
+             domain='survey' genderAnalysis={selectedChartAnalysis}
+             genderCountAnalysis={surveyCountAnalysis.genderCountAnalysis}
+             selectedResult={selectedResult}/>
+            </div> :
+            isLoaded && selectedChartAnalysis.analysisId === 3112 ?
+            <div className='chart' key='age-chart'>
+                                 <AgeChartReactComponent
+                                 domain='survey' ageAnalysis={selectedChartAnalysis}
+                                 ageCountAnalysis={surveyCountAnalysis.ageCountAnalysis}
+                                 selectedResult={selectedResult}/>
+            </div> :
+            isLoaded && selectedChartAnalysis.analysisId === 3113 ?
+            <div className='chart' key='age-chart'>
+            <VersionChartReactComponent versionAnalysis={selectedChartAnalysis}
+            surveyVersionAnalysis={versionAnalysis}
+            selectedResult={selectedResult}/>
+            </div> : null
+}
+
+      </React.Fragment>;
+  }
+}
+
+
+@Component({
+  selector: 'app-survey-chart-react',
+  template: `<span #root></span>`,
+  styleUrls: ['./survey-chart.component.css', '../../../styles/template.css', '../../../styles/page.css'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class SurveyChartWrapperComponent extends BaseReactWrapper {
+  @Input() graphButtons: string[];
+  @Input() question: any;
+  @Input() answer: any;
+  @Input() selectedAnalysis: any;
```

Comment:
Do we need this? Doesn't seem to be used in the component

Reply:
changed this thanks